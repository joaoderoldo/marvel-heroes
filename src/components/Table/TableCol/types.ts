export interface TableColProps {
  children: React.ReactNode;
  classes?: string | string[];
  dataLabel?: string;
}
