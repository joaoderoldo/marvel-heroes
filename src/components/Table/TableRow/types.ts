export interface TableRowProps {
  children: React.ReactNode;
  classes?: string | string[];
  onClick?: () => void;
}
