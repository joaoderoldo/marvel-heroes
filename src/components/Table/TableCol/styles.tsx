import styled from "styled-components";

export const TableColWrapper = styled.div`
  width: calc(100% / 3);
  display: flex;
  align-items: center;

  &:last-child {
    justify-content: flex-end;
  }

  @media all and (max-width: 767px) {
    display: flex;
    padding: 10px 0;
    width: 100%;

    &:last-child {
      justify-content: flex-start;
    }

    &:before {
      color: #6c7a89;
      padding-right: 10px;
      content: attr(data-label);
      width: 40%;
      display: flex;
      flex-shrink: 0;
    }
  }
`;
