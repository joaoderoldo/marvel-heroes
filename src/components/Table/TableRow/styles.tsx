import styled, { css } from "styled-components";

export const TableRowWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: var(--white, #fff);
  box-shadow: 0px 6px 10px 0 rgba(201, 201, 201, 1);
  border: 1px solid #c9c9c9;
  border-radius: 5px;
  padding: 16px 24px;

  ${({ clickable }) =>
    clickable &&
    css`
      cursor: pointer;
    `}

  & + & {
    margin-top: 8px;
  }

  @media all and (max-width: 767px) {
    display: block;
  }
`;
