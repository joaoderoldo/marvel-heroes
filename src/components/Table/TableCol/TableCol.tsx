import clsx from "clsx";
import { FC } from "react";

import { TableColWrapper } from "./styles";
import { TableColProps } from "./types";

const TableCol: FC<TableColProps> = ({ children, classes, dataLabel }) => {
  return (
    <TableColWrapper className={clsx(classes)} data-label={dataLabel}>
      {children}
    </TableColWrapper>
  );
};

export default TableCol;
