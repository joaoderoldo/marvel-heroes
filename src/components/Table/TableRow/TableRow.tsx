import clsx from "clsx";
import { FC } from "react";

import { TableRowWrapper } from "./styles";
import { TableRowProps } from "./types";

const TableRow: FC<TableRowProps> = ({
  children,
  classes,
  onClick,
  clickable,
}) => {
  return (
    <TableRowWrapper className={clsx(classes)} onClick={onClick}>
      {children}
    </TableRowWrapper>
  );
};

export default TableRow;
