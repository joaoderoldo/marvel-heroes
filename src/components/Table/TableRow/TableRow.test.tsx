import { vi } from "vitest";
import { render, fireEvent } from "@testing-library/react";
import TableRow from "./TableRow";
import TableCol from "../TableCol/TableCol";

interface TableHeaderMock {
  onClick: () => void;
}

const TableHeaderMock = ({ onClick }: TableHeaderMock) => (
  <TableRow classes="table-row" onClick={onClick}>
    <TableCol>Personagem</TableCol>
    <TableCol>Séries</TableCol>
    <TableCol>Eventos</TableCol>
  </TableRow>
);

describe("TableRow", () => {
  it("should render the component", () => {
    const { container } = render(
      <TableHeaderMock onClick={() => console.log("clicked!")} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render children correctly", () => {
    const { getByText } = render(
      <TableHeaderMock onClick={() => console.log("clicked!")} />
    );
    expect(getByText("Personagem")).toBeInTheDocument();
    expect(getByText("Séries")).toBeInTheDocument();
    expect(getByText("Eventos")).toBeInTheDocument();
  });

  it("should handle onClick event correctly", () => {
    const onClickMock = vi.fn();
    const { getByText } = render(<TableHeaderMock onClick={onClickMock} />);
    fireEvent.click(getByText("Personagem"));
    expect(onClickMock).toHaveBeenCalledTimes(1);
  });
});
