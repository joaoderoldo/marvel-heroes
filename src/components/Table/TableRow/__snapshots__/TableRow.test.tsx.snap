// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TableRow > should render the component 1`] = `
<div
  class="sc-beqWaB NLxfu table-row"
>
  <div
    class="sc-gueYoa fnNkuk"
  >
    Personagem
  </div>
  <div
    class="sc-gueYoa fnNkuk"
  >
    SÃ©ries
  </div>
  <div
    class="sc-gueYoa fnNkuk"
  >
    Eventos
  </div>
</div>
`;
